django/db/migrations/autodetector.py

```diff
@@ -52,21 +52,33 @@ def deep_deconstruct(self, obj):
         Used for full comparison for rename/alter; sometimes a single-level
         deconstruction will not compare correctly.
         """
-        if not hasattr(obj, 'deconstruct') or isinstance(obj, type):
-            return obj
-        deconstructed = obj.deconstruct()
-        if isinstance(obj, models.Field):
-            # we have a field which also returns a name
-            deconstructed = deconstructed[1:]
-        path, args, kwargs = deconstructed
-        return (
-            path,
-            [self.deep_deconstruct(value) for value in args],
-            {
+        if isinstance(obj, list):
+            items = [self.deep_deconstruct(value) for value in obj]
+        elif isinstance(obj, tuple):
+            items = tuple(self.deep_deconstruct(value) for value in obj)
+        elif isinstance(obj, dict):
+            items = {
                 key: self.deep_deconstruct(value)
-                for key, value in kwargs.items()
-            },
-        )
+                for key, value in obj.items()
+            }
+        elif isinstance(obj, type):
```

Comment:
Well, in that case I hope there already is a test case that'd blow up if that changes. Can you check that and, if there is no test for that, add one and also add a comment to the code why this has to be there.